modules/smitten/index.ts

```diff
@@ -6,24 +6,35 @@ import {action} from '@action-land/core'
 
 export interface Smitten<T extends string | number = string | number> {
   of<S extends string | number>(type: T): Smitten<S>
-  emit(obj: any): void
+  emit(value: any): void
+  bindTo<S extends string | number>(value: any): Smitten<S>
 }
 
 class DefaultEmitter implements Smitten {
+  private returnValue: any
+
   constructor(
     readonly type: string | number,
     readonly parent: DefaultEmitter | RootEmitter
-  ) {}
+  ) {
+    this.returnValue = null
+  }
+
+  bindTo<S extends string | number>(value: any): Smitten<S> {
+    this.returnValue = value
```

Comment:
What if I want to emit this value only for one case?

for e.g.
```
mount: () => e.emit('someValueOnlyWithMount')
unmount: e.emit // emit with whatever unmount is called
```

The current will bind the value for all emits.

Reply:
in the above case, you could do:

```js
mount: () => e.emit('mount').with('someValueForMount')
unmount: () => e.emit('unmount').with('someDifferentValue')
```

That's the behaviour you'd like, no?

Reply:
yes, this will need to be changed... will just think on this.